command/push.go

```diff
@@ -0,0 +1,301 @@
+				"for you, remote state must be used and configured. Remote\n" +
+				"state via any backend is accepted, not just Atlas. To\n" +
+				"configure remote state, use the `terraform remote config`\n" +
+				"command.")
+		return 1
+	}
+
+	// Build the context based on the arguments given
+	ctx, planned, err := c.Context(contextOpts{
+		Path:      configPath,
+		StatePath: c.Meta.statePath,
+	})
+	if err != nil {
+		c.Ui.Error(err.Error())
+		return 1
+	}
+	if planned {
+		c.Ui.Error(
+			"A plan file cannot be given as the path to the configuration.\n" +
+				"A path to a module (directory with configuration) must be given.")
+		return 1
+	}
+
+	// Get the configuration
+	config := ctx.Module().Config()
+	if name == "" {
+		if config.Atlas == nil || config.Atlas.Name == "" {
+			c.Ui.Error(
+				"The name of this Terraform configuration in Atlas must be\n" +
+					"specified within your configuration or the command-line. To\n" +
+					"set it on the command-line, use the `-name` parameter.")
+			return 1
+		}
+		name = config.Atlas.Name
+	}
+
+	// Initialize the client if it isn't given.
+	if c.client == nil {
+		// Make sure to nil out our client so our token isn't sitting around
+		defer func() { c.client = nil }()
+
+		// Initialize it to the default client, we set custom settings later
+		client := atlas.DefaultClient()
+
+		if atlasToken != "" {
+			client.Token = atlasToken
+		}
+
+		c.client = &atlasPushClient{Client: client}
+	}
+
+	// Get the variables we might already have
+	vars, err := c.client.Get(name)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf(
+			"Error looking up prior pushed configuration: %s", err))
+		return 1
+	}
+	for k, v := range vars {
+		ctx.SetVariable(k, v)
+	}
+
+	// Ask for input
+	if err := ctx.Input(c.InputMode()); err != nil {
+		c.Ui.Error(fmt.Sprintf(
+			"Error while asking for variable input:\n\n%s", err))
+		return 1
+	}
+
+	// Build the archiving options, which includes everything it can
+	// by default according to VCS rules but forcing the data directory.
+	archiveOpts := &archive.ArchiveOpts{
+		VCS: archiveVCS,
+		Extra: map[string]string{
+			DefaultDataDir: c.DataDir(),
+		},
+	}
+	if !moduleUpload {
+		// If we're not uploading modules, then exclude the modules dir.
+		archiveOpts.Exclude = append(
+			archiveOpts.Exclude,
+			filepath.Join(c.DataDir(), "modules"))
+	}
+
+	archiveR, err := archive.CreateArchive(configPath, archiveOpts)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf(
+			"An error has occurred while archiving the module for uploading:\n"+
+				"%s", err))
+		return 1
+	}
+
+	// Upsert!
+	opts := &pushUpsertOptions{
+		Name:      name,
+		Archive:   archiveR,
+		Variables: ctx.Variables(),
+	}
+	vsn, err := c.client.Upsert(opts)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf(
+			"An error occurred while uploading the module:\n\n%s", err))
+		return 1
+	}
+
+	c.Ui.Output(c.Colorize().Color(fmt.Sprintf(
+		"[reset][bold][green]Configuration %s uploaded! (v%d)",
```

Comment:
I think we should `%q` the name of the configuration so it's clearer unless you feel strongly otherwise:

```text
Configuration hashicorp/banana uploaded! (v1.0)
```

```text
Configuration "hashicorp/banana" uploaded! (v1.0)
```